src/puppetlabs/trapperkeeper/services/status/status_core.clj

```diff
@@ -52,6 +52,12 @@
 (def SemVerVersion
   (schema/pred semver/valid-format? "semver"))
 
+(def StatusProxyConfig
+  {:proxy-target-url schema/Str
```

Comment:
You can use `schema/pred` to use your `validate-protocol` function to validate the `proxy-target-url` here in the `StatusProxyConfig` schema itself, rather than separately calling your `validate-protocol` function. I'm not sure how nicely that would play with the exception you're throwing in `validate-protocol` though.